src/braket/circuits/circuit.py

```diff
@@ -877,6 +877,67 @@ def to_ir(self) -> Program:
             basis_rotation_instructions=ir_basis_rotation_instructions,
         )
 
+    @staticmethod
+    def _einsum_subscripts(targets: QubitSet, qubit_count: int) -> str:
```

Comment:
I will let @kshitijc comment on if we want the actual implementation of `as_unitary` and `_einsum_subscript` be in the circuit class or a new folder called `utils`.

Reply:
On this regard, I thought on adding `as_unitary()` as a subroutine with `@circuit.subroutine(register=True)` but given that, if we do so, we have to return `Iterable[Instruction]`, `Instruction`, `ResultType` or `Iterable[ResultType`, I discarded this idea.

Reply:
Done in https://github.com/aws/amazon-braket-sdk-python/pull/253/commits/ea7784fbe243c80d07dff84936b48b726a227e8e